sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcUtil.java

```diff
@@ -97,4 +97,23 @@ public static String generateStatement(String tableName, List<Schema.Field> fiel
         }
     }
 
+    private static JdbcIO.PreparedStatementSetCaller createBytesCaller() {
+        return (element, ps, i, fieldWithIndex) -> {
+            validateLogicalTypeLength(fieldWithIndex.getField(), element.getBytes(fieldWithIndex.getIndex()).length);
+            ps.setBytes(i + 1, element.getBytes(fieldWithIndex.getIndex()));
+        };
+    }
+
+    private static JdbcIO.PreparedStatementSetCaller createStringCaller() {
+        return (element, ps, i, fieldWithIndex) -> {
+            validateLogicalTypeLength(fieldWithIndex.getField(), element.getString(fieldWithIndex.getIndex()).length());
+            ps.setString(i + 1, element.getString(fieldWithIndex.getIndex()));
+        };
+    }
+
+    private static void validateLogicalTypeLength(Schema.Field field, Integer length) {
+        if (field.getType().getTypeName().isLogicalType()
+                && length >= Integer.valueOf(field.getType().getLogicalType().getArgument()))
```

Comment:
There's no guarantee that the logical type argument will be an integer. You can't really "validate" the data here. Instead, you should try to use the methods in the logical type to convert your base type value into a logical type value. If that fails, throw an exception.